### PR TITLE
Feature/gh service updates

### PR DESCRIPTION
**Pull request by rlarkin212**

### CODE REVIEW

File: src/server/ghService/helpers/decodingHelper.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  try {
+    parsedYaml = jsYaml.safeLoad(text);
+  } catch (error) {
+    console.log(error);
+  }
+  return parsedYaml;
+};
+
+export = {
```
Comment (Dragon1320) [inline src/server/ghService/helpers/decodingHelper.ts:44]: prefer export without the '=', might run into some weird ts shit down the line, if youre changing this, make sure you check all the other files as well cos i saw this in a few places

File: src/server/ghService/helpers/decodingHelper.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  UNICODE: "unicode",
+  // extra encoding.js type
+  AUTO: "auto",
+};
+
+const parsePackageYaml = async (buf: Buffer): Promise<string> => {
+  const detected = encodingTable[encoding.detect(buf)];
+
+  if (["utf-32", "binary", "unicode", "auto"].includes(detected)) {
+    console.log(`unsupported encoding: ${detected}`);
```
Comment (Dragon1320) [inline src/server/ghService/helpers/decodingHelper.ts:28]: probs a good idea to throw here, or handle the error somehow else cos TextDecoder will crash if passed an incompatible encoding

File: src/server/ghService/helpers/packageImageHelper.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { isNullOrUndefined } from "util";
```
Comment (Dragon1320) [inline src/server/ghService/helpers/packageImageHelper.ts:1]: just a thing i do; usually group imports into 3 - 1 for node stuff, 1 for libs, and 1 for project files in that order

File: src/server/ghService/helpers/packageImageHelper.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+import fetch from "node-fetch";
+import { URL } from "url";
+
+const getFavicon = async (url?: string): Promise<string> => {
+  if (isNullOrUndefined(url) || url === "") {
```
Comment (Dragon1320) [inline src/server/ghService/helpers/packageImageHelper.ts:7]: can use `x == null` to check for null/undefined

File: src/server/ghService/import/importPackageUtil.ts [hunk 5]
```diff
@@ -1,132 +1,75 @@
...
-      },
-    }).then((res) => res.json())),
-  );
-
-  const flatPackageFolderPaths: ManifestFolderList[] = packageFolders.flat(
-    packageFolders.length,
-  );
-  const packageFolderPaths = flatPackageFolderPaths.map((x) => x.path);
+const BASE_URL = "https://api.github.com/search/code?q=extension:yaml+repo:microsoft/winget-pkgs+path:/manifests";
+const TAKE = "100";
```
Comment (Dragon1320) [inline src/server/ghService/import/importPackageUtil.ts:11]: can this be a number? wont need the parseInt later as well

File: src/server/ghService/import/importPackageUtil.ts [hunk 6]
```diff
@@ -1,132 +1,75 @@
...
       Authorization: `token ${GITHUB_TOKEN}`,
     },
   }).then(res => res.json());
+  const totalCount = model.total_count;
+  const pageCount = Math.ceil(totalCount / parseInt(TAKE, 10));
 
-  const downloadUrl = downloadUrlPath[0].download_url;
-
-  return downloadUrl;
+  PAGECOUNT = pageCount;
```
Comment (Dragon1320) [inline src/server/ghService/import/importPackageUtil.ts:24]: probably better to return pageCount instead, youre only using it in 1 other place so youll still be making 1 call and it removes a global var

File: src/server/ghService/import/importPackageUtil.ts [hunk 7]
```diff
@@ -1,132 +1,75 @@
...
+  // set page count
+  await getPageCount();
 
-  const downloadUrlPaths: ManifestFolderList[] = await Promise.all(
-    packageFolderPaths.map((path) => fetch(`${CONTENTS_BASE_URL}/${path}`, {
+  let yamlUrls: string[] = [];
+
+  for (let page = 0; page <= PAGECOUNT; page += 1) {
+    // eslint-disable-next-line no-await-in-loop
+    const model: BatchImport = await fetch(`${BASE_URL}&page=${page}&per_page=${TAKE}`, {
```
Comment (Dragon1320) [inline src/server/ghService/import/importPackageUtil.ts:35]: pagination in the github api starts at 1

File: src/database/types/package.ts [hunk 8]
```diff
@@ -52,6 +52,7 @@ interface IPackage extends IBase {
       LicenseUrl?: string;
     }
   ];
+  Favico?: string;
```
Comment (Dragon1320) [inline src/database/types/package.ts:55]: Favico

File: src/server/ghService/helpers/decodingHelper.ts (hunk 1 above)
Comment (rlarkin212) [reply src/server/ghService/helpers/decodingHelper.ts:44]: done

File: src/server/ghService/helpers/decodingHelper.ts (hunk 2 above)
Comment (rlarkin212) [reply src/server/ghService/helpers/decodingHelper.ts:28]: done

File: src/server/ghService/helpers/packageImageHelper.ts (hunk 3 above)
Comment (rlarkin212) [reply src/server/ghService/helpers/packageImageHelper.ts:1]: done

File: src/server/ghService/helpers/packageImageHelper.ts (hunk 4 above)
Comment (rlarkin212) [reply src/server/ghService/helpers/packageImageHelper.ts:7]: done

File: src/server/ghService/import/importPackageUtil.ts (hunk 5 above)
Comment (rlarkin212) [reply src/server/ghService/import/importPackageUtil.ts:11]: done

File: src/server/ghService/import/importPackageUtil.ts (hunk 6 above)
Comment (rlarkin212) [reply src/server/ghService/import/importPackageUtil.ts:24]: done

File: src/server/ghService/import/importPackageUtil.ts (hunk 7 above)
Comment (rlarkin212) [reply src/server/ghService/import/importPackageUtil.ts:35]: done

**Pull request merged**